packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useRef } from "react";
+import { useRecoilState, useRecoilValue } from 'recoil';
+import './QRCodeDialog.scss';
+import showQRCodeDialogState from "../../atoms/showQRCodeDialogState";
+import primaryColorState from "../../atoms/primaryColorState";
+import apiKeyState from "../../atoms/apiKeyState";
+import currentLocationState from "../../atoms/currentLocationState";
+import QRCode from 'qrcode';
+import kioskLocationState from "../../atoms/kioskLocationState";
+import logoState from "../../atoms/logoState";
+import mapboxAccessTokenState from "../../atoms/mapboxAccessTokenState";
+import gmApiKeyState from "../../atoms/gmApiKeyState";
+
+/**
+ * Handle the QR Code dialog.
+ *
+ */
+function QRCodeDialog() {
+    const [, setShowQRCodeDialog] = useRecoilState(showQRCodeDialogState);
+    const elementRef = useRef()
+
+    const primaryColorProp = useRecoilValue(primaryColorState);
+    const apiKeyProp = useRecoilValue(apiKeyState);
+    const logoProp = useRecoilValue(logoState);
+    const gmApiKeyProp = useRecoilValue(gmApiKeyState);
+    const mapboxAccessTokenProp = useRecoilValue(mapboxAccessTokenState);
+
+    const directionsFrom = useRecoilValue(kioskLocationState);
+    const directionsTo = useRecoilValue(currentLocationState);
+
+    useEffect(() => {
+        if (directionsFrom && directionsTo) {
+            // The target URL when the user opens the QR code dialog
+            let targetUrl = window.location.origin;
+
+            // The interface for the existing URL search params
+            const currentParams = new URLSearchParams(window.location.search);
+
+            // The interface for the new URL search params
+            const newParams = new URLSearchParams();
+
+            /**
+             * Handle the presence of query parameters and props
+             * and append them to the newParams interface.
+             */
+            [['gmApiKey', gmApiKeyProp],
+            ['mapboxAccessToken', mapboxAccessTokenProp],
+            ['apiKey', apiKeyProp],
+            ['primaryColor', primaryColorProp],
+            ['logo', logoProp]]
+                .forEach(([queryParam, prop]) => {
+                    if (currentParams.has(queryParam)) {
+                        const queryParameter = currentParams.get(queryParam);
+                        newParams.append(queryParam, queryParameter);
+                    } else if (prop) {
+                        if (prop === primaryColorProp) {
+                            newParams.append(queryParam, primaryColorProp.replace("#", ""))
+                        } else {
+                            newParams.append(queryParam, prop)
```

Comment:
```suggestion
                            newParams.append(queryParam, prop);
```

Not a fan of semicolons anymore? 😆

Reply:
It was a test for you 🙄 To see if you pay attention